docs/_rule_template.md

```diff
@@ -1,22 +1,41 @@
+**Evaluation Context:** A string signifying at what level with the RMR data tree the rule evaluation must occur. Options are *Each Data Element* and *Data Group*. 
+- A rule that is evaluated at *Each Data Element* requires iteration over each data element that exists within a data group in the RMR. A rule assertion is then applied independently for each applicable data element. 
+- A rule that is evaluated at a *Data Group* is evaluated once for the relevant data group. This type of evaluation usually involves aggregating or calculating values up to the data group level prior to applying the assertion.  
 
 **Applicability Checks:** 
-1. <APPLICABILITY_CHECK_DESCRIPTION>
+1. This is a list of all checks that are performed to determine whether the rule is applicable to elements of the RMR. The method for conducting applicability checks is related to the *Evaluation Context*.  
+
+**Function Call:**  
+1. A list of all functions that this rule utilizes.  
 
 **Manual Checks:** <MANUAL_CHECK_DESCRIPTION>  
 
 ## Rule Logic:
-- **Applicability Check 1:** `CODE DESCRIBING APPLICABILITY CHECKS`  
-- <OTHER_LOGIC>  
-- **Rule Assertion:** `CODE DESCRIBING RULE ASSERTION`  
+- This section defines the logical evaluation of a rule using pseudocode. This description is used to develop the Rule Definition code in the RCT representing the rule.  
+    - A nested list structure should be used to denote code that operates iteratively, such as within a For Loop.
+        - Further nesting can be used, if required to iterate through data elements in the RMR file.
+- The format of a Applicability Check should follow the following:  
+**Applicability Check 1:** `(Example) if calculated_value > threshold:`  
+    - Applicability Checks should be placed within the appropriate location of the code structure. These checks should be written to affirm whether a rule is applicable to a data element or data group. More than one Applicability Checks may be used, if necessary. 
+    - The RCT will automatically handle cases that fail the Applicability Checks by outputting a *NOT_APPLICABLE* outcome. No additional description in the RDS is required for cases failing Applicability Checks.
+    - A Rule Assertion should be provided where appropriate within the code structure. The Rule Assertion should be a simple logical evaluation that is used to determine one of the four possible outcomes for a Rule Evaluation. Examples of possible Rule Assertion outcomes are: *PASS*, *FAIL*, *UNDETERMINED*    
+- A Rule Assertion section should be definined within the appropriate level of the code structure, so that the rule can be evaluated for each relevant data element in the RMR.
+- The Rule Assertion should generally follow the following format:  
+**Rule Assertion:** 
+    - `(Example) if calculated_value == expected_value: outcome == "PASS"`
+    - `(Example) elif calculated_value > expected_value: outcome == "UNDETERMINED"` 
+    - `(Example) else: outcome == "FAIL"`   
 
```

Comment:
It might be helpful to list some understood pseudo code structures such as:

For ..zone in RMR_B:
    Indent nested logic

Reply:
Makes sense. I'll add some notes on syntax.